_posts/2025-08-28-quarkus-a2a-java-0-3-0-alpha-release.adoc

```diff
@@ -0,0 +1,242 @@
+
+==== Declaring Supported Transports in the `AgentCard`
+
+The `AgentCard` is a class that describes the capabilities of an A2A server agent. Other agents or clients will use this to understand what our agent can do. Once you've added one or more dependencies on the A2A Java SDK Server Reference implementations, the next step is to add a class that creates an `AgentCard`.
+
+When creating your `AgentCard`, you must specify the primary endpoint for your A2A server agent and if your server agent supports multiple transports, these need to be specified as additional interfaces in your `AgentCard` as shown in the example below:
+
+[source,java]
+----
+import io.a2a.server.PublicAgentCard;
+import io.a2a.spec.AgentCapabilities;
+import io.a2a.spec.AgentCard;
+import io.a2a.spec.AgentSkill;
+...
+
+@ApplicationScoped
+public class WeatherAgentCardProducer {
+
+    @Produces
+    @PublicAgentCard
+    public AgentCard agentCard() {
+        return new AgentCard.Builder()
+                .name("Weather Agent")
+                .description("Helps with weather")
+                .url("http://localhost:10001") <1>
+                .preferredTransport(TransportProtocol.JSONRPC.asString()) <2>
+                .additionalInterfaces(List.of( <3>
+                        new AgentInterface(TransportProtocol.JSONRPC.asString(), "http://localhost:10001"), <4>
+                        new AgentInterface(TransportProtocol.GRPC.asString(), "localhost:9000") <5>
+                ))
+                .version("1.0.0")
+                .capabilities(new AgentCapabilities.Builder()
+                        .streaming(true)
+                        .pushNotifications(false)
+                        .stateTransitionHistory(false)
+                        .build())
+                .defaultInputModes(Collections.singletonList("text"))
+                .defaultOutputModes(Collections.singletonList("text"))
+                .skills(Collections.singletonList(new AgentSkill.Builder()
+                        .id("weather_search")
+                        .name("Search weather")
+                        .description("Helps with weather in city, or states")
+                        .tags(Collections.singletonList("weather"))
+                        .examples(List.of("weather in LA, CA"))
+                        .build()))
+                .protocolVersion("0.3.0")
+                .build();
+    }
+}
+----
+<1> The primary endpoint URL for our A2A server agent, `http://localhost:10001` in this example.
+<2> The preferred transport for our A2A server agent, `JSON-RPC` in this example. This is the transport protocol available at the primary endpoint URL.
+<3> The optional additional interfaces supported by our A2A server agent. This should include all supported transports.
+<4> The primary endpoint URL should also be specified here for completeness.
+<5> The gRPC transport URL, `localhost:9000`, indicating that our A2A server agent also supports gRPC.
```

Comment:
Was there any discussion on the public/private IP/URL issue?
Typically, a pod on kubernetes does not know its public IP (service name). So, the card cannot be published by it. Same in a container.

Reply:
Would it be possible to define a property in application.properties file that specifies the public URL using an environment variable and then the agent card could reference this property?